enterprise/cli/workspaceproxy.go

```diff
@@ -0,0 +1,506 @@
+//go:build !slim
+
+package cli
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"log"
+	"net"
+	"net/http"
+	"net/http/pprof"
+	"os/signal"
+	"regexp"
+	rpprof "runtime/pprof"
+	"time"
+
+	"golang.org/x/xerrors"
+
+	"github.com/coreos/go-systemd/daemon"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/collectors"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+
+	"github.com/coder/coder/cli"
+	"github.com/coder/coder/cli/clibase"
+	"github.com/coder/coder/cli/cliui"
+	"github.com/coder/coder/coderd/httpapi"
+	"github.com/coder/coder/coderd/httpmw"
+	"github.com/coder/coder/coderd/workspaceapps"
+	"github.com/coder/coder/codersdk"
+	"github.com/coder/coder/enterprise/wsproxy"
+)
+
+func (r *RootCmd) workspaceProxy() *clibase.Cmd {
+	cmd := &clibase.Cmd{
+		Use:     "workspace-proxy",
+		Short:   "Manage workspace proxies",
+		Aliases: []string{"proxy"},
+		Hidden:  true,
+		Handler: func(inv *clibase.Invocation) error {
+			return inv.Command.HelpHandler(inv)
+		},
+		Children: []*clibase.Cmd{
+			r.proxyServer(),
+			r.registerProxy(),
+			r.deleteProxy(),
+		},
+	}
+
+	return cmd
+}
+
+func (r *RootCmd) deleteProxy() *clibase.Cmd {
+	client := new(codersdk.Client)
+	cmd := &clibase.Cmd{
+		Use:   "delete",
+		Short: "Delete a workspace proxy",
+		Middleware: clibase.Chain(
+			clibase.RequireNArgs(1),
+			r.InitClient(client),
+		),
+		Handler: func(inv *clibase.Invocation) error {
+			ctx := inv.Context()
+			err := client.DeleteWorkspaceProxyByName(ctx, inv.Args[0])
+			if err != nil {
+				return xerrors.Errorf("delete workspace proxy %q: %w", inv.Args[0], err)
+			}
+
+			_, _ = fmt.Fprintf(inv.Stdout, "Workspace proxy %q deleted successfully\n", inv.Args[0])
+			return nil
+		},
+	}
+
+	return cmd
+}
+
+func (r *RootCmd) registerProxy() *clibase.Cmd {
```

Comment:
We should call it create TBH, the "registration" process will happen using a request from the moon with the token.